gammapy/image/core.py

```diff
@@ -1155,34 +1155,54 @@ def convolve(self, kernel, **kwargs):
         wcs = self.wcs.deepcopy() if self.wcs else None
         return self.__class__(name=self.name, data=data, wcs=wcs)
 
-    def smooth(self, kernel='gauss', width=3):
-        """Smooth the image (works on and returns a copy).
+    def smooth(self, kernel='gauss', radius=0.1 * u.deg, **kwargs):
+        """
+        Smooth the image (works on and returns a copy).
 
-        TODO: this is very preliminary and incomplete.
-        No tests yet, no control of boundary handling, ...
-        See https://github.com/gammapy/gammapy/issues/694
+        The definition of the smoothing parameter radius is equivalent to the 
+        one that is used in ds9 (see `ds9 smoothing <http://ds9.si.edu/doc/ref/how.html#Smoothing>`_).
 
         Parameters
         ----------
         kernel : {'gauss', 'disk', 'box'}
             Kernel shape
-        width : float
-            Width in pixels
+        radius : `~astropy.units.Quantity` or float
+            Smoothing width given as quantity or float. If a float is given it
+            interpreted as smoothing width in pixels. If an (angular) quantity
+            is given it converted to pixels using `SkyImage.wcs_pixel_scale()`.
+        kwargs : dict
+            Keyword arguments passed to `~scipy.ndimage.filters.uniform_filter`
+            ('box'), `~scipy.ndimage.filters.gaussian_filter` ('gauss') or
+            `~scipy.ndimage.convolve` ('disk').
 
         Returns
         -------
         image : `SkyImage`
             Smoothed image (a copy, the original object is unchanged).
         """
+        from scipy.ndimage import gaussian_filter, uniform_filter
+        from scipy.ndimage import convolve
+        from scipy.stats import gmean
+        
         image = self.copy()
 
+        if isinstance(radius, u.Quantity):
+            # use geometric mean if x an y pixel scale differ
+            radius = gmean((radius / self.wcs_pixel_scale()).value)
+        
+        if kernel in ['box', 'disk']:
```

Comment:
I would put the `width=` lines below, instead of a separate if/else block here, even if it means duplicating one simple line for box and disk. But that's a matter of taste of course, if you prefer to leave it here, OK!